internal/server/routes.go

```diff
@@ -140,41 +149,98 @@ func (s *Server) GenerateRoutes() Routes {
 type HandlerFunc[Req, Res any] func(c *gin.Context, req *Req) (Res, error)
 
 type route[Req, Res any] struct {
-	method              string
-	path                string
-	handler             HandlerFunc[Req, Res]
-	omitFromDocs        bool
-	omitFromTelemetry   bool
-	infraHeaderOptional bool
+	method                     string
+	path                       string
+	handler                    HandlerFunc[Req, Res]
+	omitFromDocs               bool
+	omitFromTelemetry          bool
+	infraVersionHeaderOptional bool
+	noAuthentication           bool
+	noOrgRequired              bool
+	requiredAuthorization      []string
+}
+
+// TODO: this is a temporary shim for demo purposes, these should be sent as
+// part of the route
+type routeGroup struct {
+	*gin.RouterGroup
+	noAuthentication bool
+	noOrgRequired    bool
 }
 
-func add[Req, Res any](a *API, r *gin.RouterGroup, route route[Req, Res]) {
+func add[Req, Res any](a *API, r *routeGroup, route route[Req, Res]) {
 	route.path = path.Join(r.BasePath(), route.path)
 
 	if !route.omitFromDocs {
 		a.register(openAPIRouteDefinition(route))
 	}
 
-	wrappedHandler := func(c *gin.Context) {
-		if !route.infraHeaderOptional {
+	// TODO: this is temporary for demo purposes
+	route.noAuthentication = r.noAuthentication
+	route.noOrgRequired = r.noOrgRequired
+
+	handler := func(c *gin.Context) {
+		if err := wrapRoute(a, route)(c); err != nil {
+			sendAPIError(c, err)
```

Comment:
With this change, we now long need to call `sendAPIError` in 2 places, here for normal requests, and in the `noRouteHandler` when we don't match a route. Previously we called it all over the middleware.

I think this cleans up the code a bunch, because now things read as regular functions that return an error, instead of:

```go
sendAPIError(c, err)
return
```